File: JavaCode/src/com/ysx/leetcode/easy/RotateString.java
package com.ysx.leetcode.easy;

/**
 * Leetcode 796. Rotate String
 * https://leetcode.com/problems/rotate-string/description/
 */
public class RotateString {

    public static void main(String[] args) {
        RotateString test = new RotateString();
        String A = "abcde";
        String B = "cdeab";
        System.out.println(test.rotateString(A, B));

        String A2 = "abcde";
        String B2 = "abced";
        System.out.println(test.rotateString(A2, B2));

    }

    /**
     * B是由A旋转得到的，所以B肯定包含在A+A的字符串中
     * @param A
     * @param B
     * @return
     */
    public boolean rotateString(String A, String B) {
        return A.length() == B.length() && (A + A).contains(B);
    }

    public boolean rotateString_old(String A, String B) {
        if (null == A || null == B || A.length() != B.length()) {
            return false;
        }
        int length = A.length();
        for (int i = 0; i < length; i++) {
            String sub1 = A.substring(0, i);
            String sub2 = A.substring(i);
            String temp = sub2 + sub1;
            if (temp.equals(B)) {
                return true;
            }
        }
        return false;
    }
}

File: KotlinCode/src/leetcode/easy/RotateString.kt
package leetcode.easy

/**
 * Leetcode 796. Rotate String
 * https://leetcode.com/problems/rotate-string/description/
 */
class RotateString{
    fun rotateString(A: String, B: String): Boolean {
        return A.length == B.length && (A + A).contains(B)
    }

    fun rotateString_old(A: String, B: String): Boolean {
        if (A.length != B.length) {
            return false
        }
        val length:Int = A.length
        for (i in 0 until length) {
            val sub1:String = A.substring(0, i)
            val sub2:String = A.substring(i)
            val temp:String = sub2 + sub1
            if (temp == B) {
                return true
            }
        }
        return false
    }
}

fun main(args: Array<String>) {
    val test = RotateString()
    val A1:String = "abcde"
    val B1:String = "cdeab"
    println(test.rotateString(A1, B1))

    val A2:String = "abcde"
    val B2:String = "abced"
    println(test.rotateString(A2, B2))
}